### PR TITLE
[Mobile] Cover: Adds Clear media button in block settings

### LINKED ISSUE #23789: [Mobile] Cover: Allow removing the selected media
**Is your feature request related to a problem? Please describe.**
The user is not able to remove an image or video after setting it

**Describe the solution you'd like**
Add a "Clear media" button in the block settings sheet under a new "Media" section

**Describe alternatives you've considered**
Add a "Clear media" option in the menu appearing when you long press on the image/video 

**Additional Context**
This issue is part of breaking down issue **https://github.com/wordpress-mobile/gutenberg-mobile/issues/2275** into smaller subtasks. Opening the new tickets here as part of the GB-Mobile Monorepo migration.

### PR DESCRIPTION
**Pull request by antonis**
**Fixes** https://github.com/WordPress/gutenberg/issues/23789 (**parent** https://github.com/wordpress-mobile/gutenberg-mobile/issues/2275)

## Related PRs:
* Gutenberg-Mobile https://github.com/wordpress-mobile/gutenberg-mobile/pull/2489
* Create cover block with only colour from a default palette https://github.com/WordPress/gutenberg/pull/23833
* Create cover block with custom colour https://github.com/WordPress/gutenberg/pull/23870
* Media Button when only colour is selected https://github.com/WordPress/gutenberg/pull/23878

## Description
Adds Clear media button in block settings

## How has this been tested?
* Add a cover block
* Press the *ADD IMAGE OR VIDEO* button and select a media
* Tap on the block settings
* Verify that there is a clear media button that removes the set image

## Screenshots

|With set media|Without set media|
|---|---|
|Simulator Screen Shot - iPhone 11 - 2020-07-08 at 18 22 53|Simulator Screen Shot - iPhone 11 - 2020-07-08 at 18 23 14|

## Types of changes
Enhancement

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by antonis** [comment]
Hello @iamthomasbishop 👋 ,
I started implementing this following your second [suggestion](https://github.com/wordpress-mobile/gutenberg-mobile/issues/2275#issuecomment-635095905).
> The other place we might want to consider showing this option is in the block settings sheet. We obviously don't have any media-related settings in the sheet yet, but I think there was some discussion about adding focal point and other items soon, so I would put this "clear media" button nearby those items (similar to web).

Please let me know if you agree with this or if I should add this to the menu appearing on long-press.

**Comment by iamthomasbishop** [comment]
@antonis looks good, just one request - can we align the "clear media" text label to the left of the cell? Normally I'd center the button like that, but it looks a bit "lonely" in this context. 😀

**Comment by antonis** [comment]
Thank you for the prompt feedback @iamthomasbishop. I updated the PR

File: packages/block-library/src/cover/edit.native.js [hunk 1]
```diff
@@ -225,6 +226,21 @@ const Cover = ( {
 					style={ styles.rangeCellContainer }
 				/>
 			</PanelBody>
+
+			{ url ? (
+				<PanelBody title={ __( 'Media' ) }>
+					<BottomSheetButton
```
Comment (geriux) [inline packages/block-library/src/cover/edit.native.js:232]: For this case we could just import `BottomSheet` which exports pretty much all components that it uses included the `Button` something like

`import { BottomSheet } from '@wordpress/components';`

And then you can use `BottomSheet.Button` 

For this issue, I think we could go with `BottomSheet.Cell` 

Something like:

```
<BottomSheet.Cell
	leftAlign
	label={ __( 'Clear Media' ) }
	labelStyle={ styles.clearMedia }
	onPress={ () => {
		setAttributes( { id: undefined, url: undefined } );
	} }
/>
```

What do you think?

**Comment by geriux** [comment]
Hey @antonis 👋 

While testing I think I found a small bug, when I tap on Clear media it clears the media successfully but the bottom sheet remains open:

<kbd></kbd>

**Comment by antonis** [comment]
Thank you for your feedback @geriux 🙇 
I agree with your suggestions. I submitted the changes along with the bug fix leaving the bottom sheet open.

File: packages/components/src/mobile/bottom-sheet/button.native.js [hunk 2]
```diff
@@ -8,14 +8,28 @@ import { TouchableOpacity, View, Text } from 'react-native';
  */
 import styles from './styles.scss';
 
-const BottomSheetButton = ( { onPress, disabled, text, color } ) => (
+const BottomSheetButton = ( {
+	onPress,
+	disabled,
+	text,
+	color,
+	justifyContent,
```
Comment (geriux) [inline packages/components/src/mobile/bottom-sheet/button.native.js:16]: I don't think we need these changes after the latest update right?

Comment (antonis) [reply packages/components/src/mobile/bottom-sheet/button.native.js]: You are right @geriux. I removed the unneeded BottomSheetButton customisation.

**Review by geriux — APPROVED** [review]
LGTM! Nice work @antonis! 🚀 Tested it on both iOS and Android.

**Comment by iamthomasbishop** [comment]
On my way.

**Pull request merged**